src/Stratis.Features.SystemContracts/Contracts/DataStorageContract.cs

```diff
@@ -116,6 +121,15 @@ public Result<object> Dispatch(ISystemContractTransactionContext context)
 
                         return Result.Fail<object>($"Method {context.CallData.MethodName} overload with {context.CallData.Parameters.Length} params does not exist on type {nameof(DataStorageContract)} v{context.CallData.Version}");
 
+                    case nameof(DataStorageContract.PersistComplexType):
+                        var txRawHex = context.CallData.Parameters[0] as string;
+
+                        var tx = Transaction.Parse(txRawHex, RawFormat.Satoshi);
+
+                        instance.PersistComplexType(tx);
```

Comment:
Let's just pass complex types the same way we would for smart contracts - i.e. by serializing them to byte arrays. There is no requirement to support complex types on public methods.

Reply:
This is just an example, when implementing it you can choose never to do this.